### ISSUE TITLE
Resubscribing a customer from Magento throws a 500 error due to Fatal Uncaught Error: "Call to a member function getStreetLine() on string"

### ISSUE THREAD

**Issue by jordanbrauer** [mailchimp/mc-magento2#62]
## Preconditions:

1. Magento CE `2.1.5`
2. MailChimp for Magento 2 `1.0.21`
3. PHP 7

## Steps to reproduce:

1. Install MailChimp for Magento 2 with `composer require mailchimp/mc-magento2 1.0.21`.
2. Configure MailChimp for Magento 2 following [this guide](http://kb.mailchimp.com/integrations/e-commerce/connect-or-disconnect-mailchimp-for-magento-2).
3. Ensure that the "Enable Two Way Sync" option is enabled.
4. Wait for the Magento customers to sync to the assigned MailChimp list.
5. Unsubscribe a customer from the newsletter in Magento from either the admin or customer dashboard.
6. Check the same customers' "Email Marketing" status in MailChimp to ensure they were unsubscribed (This checks out and works just fine)
7. Return to Magento and attempt to resubscribe the same customer to the newsletter from either the admin or customer dashboard (try both).

## Expected result:

* The page reloads with a successful `200`/`300` response code.
* The customer is resubscribed to the newsletter and a UI notification indicating so.  

**Customer Dashboard:** 
![image](https://user-images.githubusercontent.com/18744334/29182217-afb14796-7dc3-11e7-8168-ee130c7a7706.png)

**Admin Dashboard:** 
![image](https://user-images.githubusercontent.com/18744334/29182197-9649e51a-7dc3-11e7-9273-9ee40cb8e7a1.png)

## Actual result:

* The page reloads with a `500` error code
* The customer is left unsubscribed to thew newsletter and no UI notification.

**Admin & Customer Dashboard response pages (+ newsletter footer signup form):** ![image](https://user-images.githubusercontent.com/18744334/29182180-85001450-7dc3-11e7-9a4a-25e2fa7ec16d.png)

**Error stack trace:**
```
Fatal error: Uncaught Error: Call to a member function getStreetLine() on string in
{{ magento_dir }}/vendor/mailchimp/mc-magento2/Helper/Data.php:533

Stack trace:

#0 {{ magento_dir }}/vendor/mailchimp/mc-magento2/Helper/Data.php(435):
Ebizmarts\MailChimp\Helper\Data->_updateMergeVars('BILLING', 'Billing', Object(Magento\Customer\Model\Data\Customer))
#1 {{ magento_dir }}/vendor/mailchimp/mc-magento2/Helper/Data.php(395):
Ebizmarts\MailChimp\Helper\Data->_getCustomerMergeVarsValues(Array, Object(Magento\Customer\Model\Data\Customer), Array)
#2 {{ magento_dir }}/vendor/mailchimp/mc-magento2/Model/Plugin/Subscriber.php(123):
Ebizmarts\MailChimp\Helper\Data->getMergeVars(Object(Magento\Newsletter\Model\Subscriber\Interceptor), 'customer@email...')
#3 {{ magento_dir }}/vendor/magento/framework/Interception/Interceptor.php(123):
in {{ magento_dir }}/vendor/mailchimp/mc-magento2/Helper/Data.php on line 533
```
_Notes (stack trace):_
`{{ magento_dir }}` = _installation directory of Magento 2_
`customer@email` = _the customer in questions' email address_

## Additional information:

~_N/A for now..._~

I found that while attempting to resubscribe to the newsletter via the customer dashboard, the dev tool console shows a `404` for `https://{{ site_domain }}/newsletter/manage/mailchimp/script/get` (see screen shot below)

![image](https://user-images.githubusercontent.com/18744334/29185169-18a480c4-7dce-11e7-9596-197847f077b1.png)

**Comment by gonzaloebiz** [comment]
Hi @jordanbrauer 

thanks for your feedback.
I follow your steps with the last version of the extension (1.0.22) and works for me.
Can you update the extension and try yourself?

For now i close the issue, if you confirm the issue with the last version, i will re open the issue.

Best

**Comment by jordanbrauer** [comment]
Thanks for testing it out @gonzaloebiz!

I upgraded to `1.0.22` and still have the same issue unfortunately. I think it might have something to with the `404` in update i added to my Additional Information section.

**Comment by gonzaloebiz** [comment]
So weird, i try and works for me.
You have any extension that modify the customer?
By the way, can you try after apply this https://github.com/mailchimp/mc-magento2/commit/c7967fb1ed05fb81429b37df2cf6525215a037e2 ? It's a fix for the ajax call issue

Best

**Comment by jordanbrauer** [comment]
That is strange. I will have to spin a new store up and try it there too and see what happens.

Not that I am aware of - nothing intentional at least. We've never needed to add features or make changes to the way customers are managed, especially the newsletter.

I'll give that a go right now for the AJAX fix and report back the results.

**Comment by jordanbrauer** [comment]
@gonzaloebiz The `404` AJAX issue was fixed by your update. The `500` error from attempting a resubscribe to the newsletter however, was not. 😞 

Any idea as to what could be happening from what callstack that I showed? In the mean time, I will dig through my Magento logs and post any relevant info.

**Comment by gonzaloebiz** [comment]
So weird.
You have the webserver log? (apache or nginx) If you have a 500 for sure something in the log can help us

Best

**Comment by jordanbrauer** [comment]
Here's what I could find in the server logs available to us.

**Apache Log**

```log
SERVER.IP.ADDRESS - - [11/Aug/2017:12:09:41 -0400] "POST /newsletter/manage/save/ HTTP/1.1" 500 - "https://{{ site_domain }}/newsletter/manage/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36"
```

Doesn't seem to give much other than the source, timestamp, method, endpoint, response code, and user agent info.

**PHP Error Log**

```log
Stack trace:
[11-Aug-2017 16:01:07 UTC] PHP Fatal error:  Uncaught Error: Call to a member function getStreetLine() on string in {{ magento_dir }}/vendor/mailchimp/mc-magento2/Helper/Data.php:533
#3 {{ magento_dir }}/vendor/magento/framework/Interception/Interceptor.php(123): Ebizmarts\Ma in {{ magento_dir }}/vendor/mailchimp/mc-magento2/Helper/Data.php on line 533
#2 {{ magento_dir }}/vendor/mailchimp/mc-magento2/Model/Plugin/Subscriber.php(83): Ebizmarts\MailChimp\Helper\Data->getMergeVars(Object(Magento\Customer\Model\Data\Customer), 'customer@email...')
#1 {{ magento_dir }}/vendor/mailchimp/mc-magento2/Helper/Data.php(395): Ebizmarts\MailChimp\Helper\Data->_getCustomerMergeVarsValues(Array, Object(Magento\Customer\Model\Data\Customer), Array)
#0 {{ magento_dir }}/vendor/mailchimp/mc-magento2/Helper/Data.php(435): Ebizmarts\MailChimp\Helper\Data->_updateMergeVars('BILLING', 'Billing', Object(Magento\Customer\Model\Data\Customer))
```

The PHP error log is the same as what the stack trace that the `500` error outputs to the browser with `ini_set('display_errors', 1);` in `{{ magento_dir }}/app/bootstrap.php`

What's weird is that this issue only seems to happen on our production server. I am not sure if that has anything to do with the fact that we are not syncing any lists to our development and staging stores though (to prevent duplication of our products in the MailChimp content manager).

---

I also tried resubbing the customer to the newsletter from the admin customer grid with like so,

![image](https://user-images.githubusercontent.com/18744334/29222209-335b7436-7e87-11e7-8deb-098adcd1f621.png)

but received the following error _after the `500` error and refreshing the page_

![image](https://user-images.githubusercontent.com/18744334/29222244-4f467a74-7e87-11e7-8a2f-07b617dc6006.png)

**Comment by gonzaloebiz** [comment]
Hi @jordanbrauer 

finally i can reproduce your issue.
Take a look at the associated commit and let me know if works for you.

Best

**Comment by jordanbrauer** [comment]
Fantastic! It works as expected now. 😃

Thanks a bunch @gonzaloebiz.

Can we make this and the AJAX call fix you committed earlier a `1.0.23` release? Currently I've got patch files that are applied during our deployment process, but it would be nice to drop those. :)
